controller/array_action/array_mediator_svc.py

```diff
@@ -294,6 +297,53 @@ def get_host_by_host_identifiers(self, initiators):
             logger.debug("can not found host by using initiators: {0} ".format(initiators))
             raise controller_errors.HostNotFoundError(initiators)
 
+    def _get_detailed_hosts_list(self):
+        logger.debug("Getting detailed hosts list on array {0}".format(self.endpoint))
+        hosts_list = self.client.svcinfo.lshost()
+        if not hosts_list:
+            return []
+
+        # get all hosts details by sending a single batch of commands, in which each command is per host
+        detailed_hosts_list_cmd = self._get_detailed_hosts_list_cmd(hosts_list)
+        logger.debug("Sending getting detailed hosts list commands batch")
+        detailed_hosts_list_output, detailed_hosts_list_errors = self._send_raw_cli_command(detailed_hosts_list_cmd)
+        if detailed_hosts_list_errors:
+            logger.error("Errors returned from getting detailed hosts list: {0}".format(detailed_hosts_list_errors))
+
+        logger.debug("Reading detailed hosts list commands batch response")
+        hosts_reader = SVCListResultsReader(detailed_hosts_list_output)
+        res = []
+        for host_details in hosts_reader:
+            host_id = host_details.get(HOST_ID_PARAM)
+            host_name = host_details.get(HOST_NAME_PARAM)
+            iscsi_names = host_details.get_as_list(HOST_ISCSI_NAMES_PARAM)
+            wwns = host_details.get_as_list(HOST_WWPNS_PARAM)
+            host = Host(host_id, host_name, iscsi_names, wwns)
```

Comment:
opening the Host class vs Munch issue here, since the [conversation with numbered issues](https://github.com/IBM/ibm-block-csi-driver/pull/154/files#r421501818) is not convenient.
as I wrote there: could you please point me to the "get_hosts_for_init" ticket?

regardless of this ticket, I still think a Munch will save us the new consts, and conforms to the project's current convention for cli objects.

Reply:
consider here CSI-1492